### ISSUE TITLE
DS client and connection properties 

### ISSUE THREAD

**Issue by jfgosselin** [Impetus/kundera#555]
It seems that I can only pass the DS connection properties via a kundera.client.property file.  Isn't there a way to pass these properties in a Map ? 

From DSClientFactory the properties are only read via the file :
     Properties connectionProperties = CassandraPropertyReader.csmd.getConnectionProperties();)

```
protected Object createPoolOrConnection()
{
    // AuthInfoProvider,LoadBalancingPolicy, ReconnectionPolicy,
    // RetryPolicy,ProtocolOptions.Compression, SSLOptions,
    // PoolingOptions,SocketOptions

    if (logger.isDebugEnabled())
    {
        logger.debug("Intiatilzing connection");
    }

    Properties connectionProperties = CassandraPropertyReader.csmd.getConnectionProperties();
```

Thanks

JF

**Comment by mevivs** [comment]
Will fix it.Need to cross check with other clients.

-Vivek

**Comment by mevivs** [comment]
Not an issue. 
Properties which can be overriden other that file are host, port and keyspace. which is also valid in case of DS client.  
=>
https://github.com/impetus-opensource/Kundera/blob/trunk/src/jpa-engine/core/src/main/java/com/impetus/kundera/service/HostConfiguration.java#L78

-Vivek

**Comment by jfgosselin** [comment]
How can I set e.g. loadbalancing.policy without a kundera.client.property file ?

**Comment by mevivs** [comment]
Not allowed. Such client specific properties are only allowed via external configuration file.

-Vivek

**Comment by jfgosselin** [comment]
The section "External Properties Configuration" in the wiki wasn't clear to me.
"DS specific properties can be set externally via property map as well."
https://github.com/impetus-opensource/Kundera/wiki/Datastax-java-driver-support-for-Cassandra-using-Kundera

Enhancement ? 

Thanks

JF

**Comment by mevivs** [comment]
Will update wiki. Only host, port an keyspace can be overridden via external map.

-Vivek

**Comment by impetus-opensource-admin** [comment]
Closed. Released with 2.11.1
